nav2_bt_navigator/src/bt_navigator.cpp

```diff
@@ -168,7 +168,19 @@ BtNavigator::navigateToPose()
 {
   initializeGoalPose();
 
-  auto is_canceling = [this]() {return action_server_->is_cancel_requested();};
+  auto is_canceling = [this]() {
+      if (action_server_ == nullptr) {
```

Comment:
This check should probably be done once when getting into the callback.

Reply:
We have to check for nullptr because if the cleanup() is called (resetting the action server shared pointer) and the thread calling this function hasn't stopped already, we would run into a segmentation fault.